### ISSUE TITLE
py.test has lost the nice assertion failure display

### ISSUE THREAD

**Issue by ThomasWaldmann** [borgbackup/borg#1938]
when fixing #1930, I noticed that when the assertion fails, it just says "AssertionError", it does not display the values of the terms in the assertion expression.

**Comment by ThomasWaldmann** [comment]
E.g. like here:
```
__________________________________________________________ TestLocationWithoutEnv.test_file ___________________________________________________________

self = <borg.testsuite.helpers.TestLocationWithoutEnv object at 0x7fced6086940>

    def test_file(self):
        #monkeypatch.delenv('BORG_REPO', raising=False)
        assert repr(Location('xfile:///some/path::archive')) == \
>           "Location(proto='file', user=None, host=None, port=None, path='/some/path', archive='archive')"
E       AssertionError

borg/testsuite/helpers.py:48: AssertionError
```
The lhs value would have been interesting...

**Comment by RonnyPfannschmidt** [comment]
i believe with the particular layout for the helpers, you need 

http://doc.pytest.org/en/latest/writing_plugins.html#pytest.register_assert_rewrite

**Comment by ThomasWaldmann** [comment]
fixed by #1951.

**Comment by ThomasWaldmann** [comment]
Reopening, see comments in #1951.

**Comment by ThomasWaldmann** [comment]
Fixed by #1959.
